backend/server/routers/utility.py

```diff
@@ -32,3 +37,13 @@ def get_core_courses(program: str, specialisations: list[str]):
             if "Major" in spec_name or "Honours" in spec_name)
          , [])
 
+
+def get_course(code: str) -> Course:
```

Comment:
I think we can collect this data already using `getTermsOffered` and `getCourse` from our own apis

Reply:
Yup `getTermsOffered` would be helpful.
But not sure which `getCourse` you mean as `get_course` in `courses.py` returns a Dict with stuff like `is_legacy`, `handbook_note`, which is not helpful to convert to `Course` i think.

Reply:
Oh yea you already got the uoc some other way nvm. Then getTermsOffered is what we need to call to make this accurate. We also need the FE to give us their mark.